diff --git a/flash_attn/cute/flash_fwd_sm100_fp4.py b/flash_attn/cute/flash_fwd_sm100_fp4.py
index 5705db2..e3f233a 100644
--- a/flash_attn/cute/flash_fwd_sm100_fp4.py
+++ b/flash_attn/cute/flash_fwd_sm100_fp4.py
@@ -3929,6 +3929,7 @@ class FlashAttentionForwardSm100:
         softmax_corr_consumer_phase = Int32(0)
         o_corr_consumer_phase = Int32(0)
         corr_epi_producer_phase = Int32(1)
+        fp8_pv_p_log2_offset = Float32(0.0)
 
         tile_scheduler = TileSchedulerCls()
         work_tile = tile_scheduler.initial_work_tile_info()
